src/main/java/com/Elessar/app/server/MyServer.java

```diff
@@ -21,27 +26,40 @@
     private final String serverName;
     private final int port;
     private final MyDatabase db;
+    private final OperationsMonitor monitor;
+    private static Timer timer;
 
     public MyServer(String serverName, int port, MyDatabase db) {
         this.serverName = serverName;
         this.port = port;
         this.db = db;
+        this.monitor = new OperationsMonitor();
+        timer = new Timer();
     }
 
     public void run() {
         try {
             final HttpServer server = HttpServer.create(new InetSocketAddress(serverName, port), 0);
             server.createContext("/", new RootHandler());
             server.createContext("/echo", new EchoHandler());
-            server.createContext("/register", new RegisterHandler(db));
-            server.createContext("/logon", new LogOnHandler(db));
-            server.createContext("/logoff", new LogOffHandler(db));
+            server.createContext("/register", new RegisterHandler(db, monitor));
+            server.createContext("/logon", new LogOnHandler(db, monitor));
+            server.createContext("/logoff", new LogOffHandler(db, monitor));
             server.setExecutor(null);
             server.start();
             logger.info("Server started at port {}", port);
         } catch (IOException e) {
             logger.fatal("Caught exception during server startup: {}", e);
         }
+        timer.scheduleAtFixedRate(new TimerTask() {
```

Comment:
This would work but with a catch. If we get a lot of requests coming in during the fixed interval then we can end up using much more memory than we want.
A better way is probably to log the metrics after we collect a fixed number of data points so the memory usage is roughly constant.